src/packagedcode/rubygems.py

```diff
@@ -647,93 +648,135 @@ def normalize(gem_data, known_fields=known_fields):
     )
 
 
-def parse_spec(location):
-    pass
+# defaults fields of .gemspec file
+data_dic = {
+    "name": None,
+    "version": None,
+    "authors": None,
+    "email": None,
+    "summary": None,
+    "description": None,
+    "homepage": None,
+    "license": None,
+    "dependencies": None
+    }
 
 
-class GemSpec(object):
+def parse_spec(location):
+    """
+    Returns a dictionary which containing .gemspec file data.
+    """
+    file_data = []
+    with io.open(location, "r", encoding="utf-8") as f:
+        file_data = [line.rstrip('\n') for line in f]
```

Comment:
Is it necessary to perform the `rstrip` for every line? Would `file_data = f.read()` work just as well?

Reply:
@JonoYang What I am actually doing is that I am splitting it line by line and store in a list. `file_data = f.read()` will not give the desired output. 

The desired output can be obtained by 2 solutions:
```python
with io.open(file_name, "r", encoding="utf-8") as f:
    file_data = [line.rstrip() for line in f]
```
or
```python
with io.open(file_name, "r", encoding="utf-8") as f:
    file_data = f.read().splitlines()
```

Please tell me which will be the better approach for this??

Reply:
@rpotter12 Thanks for clearing that up. I was mistaken initially.

I think you could iterate through the file line by line doing this:

```
with io.open(file_name, "r", encoding="utf-8") as f:
    for line in f:
        <process line>
```

Though what you have works, I do not see a need for change.

Reply:
@rpotter12 Does `.splitlines()` return a list of lines? if so, this is more readable to me IMO.